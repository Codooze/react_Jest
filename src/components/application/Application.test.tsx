import { render } from "@testing-library/react";
import { Application } from "./application";

describe("Application", () => {
  it("should render correctly", () => {
    const { container } = render(<Application />);
    expect(container).toMatchSnapshot();
  });

  //checking a specific element in the DOM
  it("should render the input element", () => {
    const { getByLabelText } = render(<Application />);
    const inputElement = getByLabelText("Name");
    expect(inputElement).toBeInTheDocument();

    //checking the value of the input element
    expect(inputElement).toHaveValue("Vishwas");
  });

  //getbyAltText
  it("should render the image element", () => {
    const { getByAltText } = render(<Application />);
    const imageElement = getByAltText("a person with a laptop");
    expect(imageElement).toBeInTheDocument();
  });

  //getbyTestId
  it("should render the custom element", () => {
    const { getByTestId } = render(<Application />);
    const customElement = getByTestId("custom-element");
    expect(customElement).toBeInTheDocument();
  });
});
