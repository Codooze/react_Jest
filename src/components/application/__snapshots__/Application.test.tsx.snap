// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application should render correctly 1`] = `
<div>
  <h1>
    Job application form
  </h1>
  <h2>
    Section 1
  </h2>
  <p>
    All fields are mandatory
  </p>
  <span
    title="close"
  >
    X
  </span>
  <img
    alt="a person with a laptop"
    src="https://via.placeholder.com/150"
  />
  <div
    data-testid="custom-element"
  >
    Custom HTML element
  </div>
  <form>
    <div>
      <label
        for="name"
      >
        Name
      </label>
      <input
        id="name"
        placeholder="Fullname"
        type="text"
        value="Vishwas"
      />
    </div>
    <div>
      <label
        for="bio"
      >
        Bio
      </label>
      <textarea
        id="bio"
      />
    </div>
    <div>
      <label
        for="job-location"
      >
        Job location
      </label>
      <select
        id="job-location"
      >
        <option
          value=""
        >
          Select a country
        </option>
        <option
          value="US"
        >
          United States
        </option>
        <option
          value="GB"
        >
          United Kingdom
        </option>
        <option
          value="CA"
        >
          Canada
        </option>
        <option
          value="IN"
        >
          India
        </option>
        <option
          value="AU"
        >
          Australia
        </option>
      </select>
    </div>
    <div>
      <label>
        <input
          id="terms"
          type="checkbox"
        />
         I agree to the terms and conditions
      </label>
    </div>
    <button
      disabled=""
    >
      Submit
    </button>
  </form>
</div>
`;
